engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/CallerMethodVisitor.java

```diff
@@ -29,6 +31,21 @@ public void visitINVOKEVIRTUAL(INVOKEVIRTUAL i) {
                 new Vertex.Builder(i.getReferenceType(cp).toString(),
                         formatMethod(i.getMethodName(cp), i.getArgumentTypes(cp))).build(),
                 LinkType.METHOD_CALL));
+        handleReflectionCall(i);
+    }
+
+    private void handleReflectionCall(InvokeInstruction i) {
+        if(isReflectionCall(i)) {
+            addLink(LinkFactory.createInstance(
+                    context.getApplicationId(),
+                    currentMethodVertex,
+                    REFLECTION_CALL_VERTEX,
+                    LinkType.REFLECTION_CALL));
+        }
+    }
+
+    private boolean isReflectionCall(InvokeInstruction i) {
+        return i.getClassName(cp).matches(REGEX_REFLECT);
```

Comment:
as the regex is very simple, what performs better? the .matches  or .contains?

Reply:
I will check